### DISCUSSION TITLE
Fast piano dispersion Filter questions

### DISCUSSION THREAD

**Discussion by Reaper10** [grame-cncm/faust#968] [Q&A]
The first question I have How to do with a piano lid? After almost pianos have a lid to them and opening or slightly opening the lid or slightly closing it changes Is that function going to be added to that library Function
https://faustlibraries.grame.fr/libs/misceffects/#efpiano_dispersion_filter

My second question has to do with making piano like instrumedisturbance using this filter To be clearer what I mean by piano Like instruments Are instruments like [Prepared piano](https://en.wikipedia.org/wiki/Prepared_piano), [harpsichord](https://en.wikipedia.org/wiki/Harpsichord), [Clavichord](https://en.wikipedia.org/wiki/Clavichord), [Celesta ](https://en.wikipedia.org/wiki/Celesta). Or would this require a new variation of the piano dispersion filter?

The third question has to do with the instruments like an upright bass Would you have to and modeling one of these instruments Make a new body being that it's technically not a guicar. So you can't really use a standard guitar body. Cause an upright base is a lot bigger than a standard  Guitar.

The reason why I asked would be good to be able to build a physical modeling synthesizer That you could make synthesizers sound like these instruments for those of us that can't afford A multi sample library one of the close source physical modeling synthesizers or a multi sample library, one of the close source physical modeling synthesizers.

**Comment by DBraun** [comment]
Faust has a [piano.dsp](https://github.com/grame-cncm/faust/blob/master-dev/examples/physicalModeling/faust-stk/piano.dsp) example but I had to adapt it to run in the online IDE. You should be able to copy paste this code into the IDE, set polyphony to 4 or so and then play notes with the keyboard.

https://gist.github.com/DBraun/3d1c735ffb414f7ce371b28a20559e30

There is a lot to explore with the [physical modeling](https://faustlibraries.grame.fr/libs/physmodels/) library. You may also want to refer to Julius Smith's [PASP](https://ccrma.stanford.edu/~jos/pasp/).
